labonneboite/scripts/create_index.py

```diff
@@ -396,56 +396,61 @@ def get_scores_by_rome(office, office_to_update=None):
     office_nafs = [office.naf]
     # Handle NAFs added to a company
     if office_to_update:
-        office_nafs += list(office_to_update.as_list(office_to_update.nafs_to_add))
+        office_nafs += office_to_update.as_list(office_to_update.nafs_to_add)
 
+    romes_to_boost = romes_to_remove = []
+    if office_to_update:
+        romes_to_boost = office_to_update.as_list(office_to_update.romes_to_boost)
+        romes_to_remove = office_to_update.as_list(office_to_update.romes_to_remove)
 
-    rome_codes = []
     for naf in office_nafs:
         try:
-            rome_codes += mapping_util.get_romes_for_naf(naf)
+            rome_codes = mapping_util.get_romes_for_naf(naf)
         except KeyError:
             # unfortunately some NAF codes have no matching ROME at all
-            pass
+            continue
 
-    romes_to_boost = []
-    if office_to_update:
         # Add unrelated rome for indexing (with boost)
-        romes_to_boost = office_to_update.as_list(office_to_update.romes_to_boost)
         rome_codes += list(set(romes_to_boost) - set(rome_codes))
 
         # Remove unwanted romes
-        romes_to_remove = office_to_update.as_list(office_to_update.romes_to_remove)
         rome_codes = list(set(rome_codes) - set(romes_to_remove))
```

Comment:
Je me suis fait des nœuds au cerveau pour comprendre. En fait on peut remplacer ces deux lignes par : 

    rome_codes = list(set(rome_codes) + set(romes_to_boost) - set(romes_to_remove))

Et en fait la conversion en liste ne sert à rien vu qu'on va juste itérer dessus. Donc on peut écrire : 

    rome_codes = set(rome_codes) + set(romes_to_boost) - set(romes_to_remove)

non ?